module/VuFind/src/VuFind/Cover/Loader.php

```diff
@@ -672,4 +656,60 @@ protected function processImageURL($url, $cache = true)
             return true;
         }
     }
+
+    /**
+     * Get urls for defined provider, works as generator
+     *
+     * @return array
+     */
+    public function getCoverUrls()
+    {
+        $ids = $this->getIdentifiers();
+        $handlers = $this->getHandlers();
+        foreach ($handlers as $handler) {
+            try {
+                // Is the current provider appropriate for the available data?
+                if ($handler['handler']->supports($ids)) {
+                    $url = $handler['handler']
+                        ->getUrl($handler['key'], $this->size, $ids);
+                    if ($url) {
+                        yield [
+                            'url' => $url,
+                            'apiName' => $handler['apiName'],
+                            'handler' => $handler['handler'],
+                        ];
+                    }
+                }
+            } catch (\Exception $e) {
+                $this->debug(
+                    get_class($e) . ' during processing of ' . $apiName
```

Comment:
`$apiName` is not defined; should be `$handler['apiName']`.

Reply:
Right, I'll fix it